tests/legacy-cli/e2e/tests/i18n/ivy-localize-serviceworker.ts

```diff
@@ -0,0 +1,189 @@
+      }
+      serveConfigs[lang] = { browserTarget: `test-project:build:${lang}` };
+      e2eConfigs[lang] = {
+        specs: [`./src/app.${lang}.e2e-spec.ts`],
+        devServerTarget: `test-project:serve:${lang}`,
+      };
+    }
+  });
+
+  // Add service worker source configuration
+  const manifest = {
+    index: '/index.html',
+    assetGroups: [
+      {
+        name: 'app',
+        installMode: 'prefetch',
+        resources: {
+          files: [
+            '/favicon.ico',
+            '/index.html',
+            '/*.bundle.css',
+            '/*.bundle.js',
+            '/*.chunk.js',
+          ],
+        },
+      },
+      {
+        name: 'assets',
+        installMode: 'lazy',
+        updateMode: 'prefetch',
+        resources: {
+          files: [
+            '/assets/**',
+            '/*.(eot|svg|cur|jpg|png|webp|gif|otf|ttf|woff|woff2|ani)',
+          ],
+        },
+      },
+    ],
+  };
+  await writeFile('ngsw-config.json', JSON.stringify(manifest));
+
+  // Add a translatable element.
+  await writeFile(
+    'src/app/app.component.html',
+    '<h1 i18n="An introduction header for this sample">Hello i18n!</h1>',
+  );
+
+  // Extract the translation messages and copy them for each language.
+  await ng('xi18n', '--output-path=src/locale');
+  await expectFileToExist('src/locale/messages.xlf');
+  await expectFileToMatch('src/locale/messages.xlf', `source-language="en-US"`);
+  await expectFileToMatch('src/locale/messages.xlf', `An introduction header for this sample`);
+
+  for (const { lang, translation } of langTranslations) {
+    if (lang != 'en-US') {
+      await copyFile('src/locale/messages.xlf', `src/locale/messages.${lang}.xlf`);
+      await replaceInFile(
+        `src/locale/messages.${lang}.xlf`,
+        'source-language="en-US"',
+        `source-language="en-US" target-language="${lang}"`,
+      );
+      await replaceInFile(
+        `src/locale/messages.${lang}.xlf`,
+        '<source>Hello i18n!</source>',
+        `<source>Hello i18n!</source>\n<target>${translation}</target>`,
+      );
+    }
+  }
+
+  // Build each locale and verify the output.
+  await ng('build', '--i18n-missing-translation', 'error');
+  for (const { lang, translation } of langTranslations) {
+    await expectFileToMatch(`${baseDir}/${lang}/main-es5.js`, translation);
+    await expectFileToMatch(`${baseDir}/${lang}/main-es2015.js`, translation);
+    await expectToFail(() => expectFileToMatch(`${baseDir}/${lang}/main-es5.js`, '$localize`'));
+    await expectToFail(() => expectFileToMatch(`${baseDir}/${lang}/main-es2015.js`, '$localize`'));
+    await expectFileToMatch(`${baseDir}/${lang}/main-es5.js`, lang);
+    await expectFileToMatch(`${baseDir}/${lang}/main-es2015.js`, lang);
+
+    // Expect service worker configuration to be present
+    await expectFileToExist(`${baseDir}/${lang}/ngsw.json`);
+
+    // Ivy i18n doesn't yet work with `ng serve` so we must use a separate server.
+    const app = express();
```

Comment:
From here downwards is redundant, since we are not testing that the service worker works.

Reply:
My goal there was to ensure that the service worker augmentation didn't break the application.